### PR TITLE
Refactoring PrivateDataManager.js and add ability of removing site data

### PR DESCRIPTION
**Pull request by atatarov**
Use a new API for clearing cookie and cache.
Replace clearing of "cookie" with "cookies-and-site-data"

### CODE REVIEW

**Review by okodron — APPROVED** [review]

File: jscomps/PrivateDataManager.js [hunk 1]
```diff
@@ -54,13 +43,15 @@ PrivateDataManager.prototype = {
...
+        this.clearData(Ci.nsIClearDataService.CLEAR_COOKIES);
         break;
       }
       case "cache": {
-        this.clearCaches()
-        debug("Cache cleaned");
+        this.clearData(Ci.nsIClearDataService.CLEAR_ALL_CACHES);
+        break;
+      }
+      case "site-data": {
```
Comment (rainemak) [inline jscomps/PrivateDataManager.js:53]: At embedlite-components side maybe it's ok to keep "cookies" and "site data" separated but in the user interface I think we should combine these. The "site data" is not very clear for end user.

Comment (okodron) [reply jscomps/PrivateDataManager.js:53]: Let's clear cookies and site data at once as firefox does it.

File: jscomps/PrivateDataManager.js [hunk 2]
```diff
@@ -26,24 +26,13 @@ PrivateDataManager.prototype = {
...
-    try {
-      imageCache.clearCache(false); // true=chrome, false=content
-    } catch(e) {
-      debug("error in clearing image cache: " + e);
-    }
+  clearData: function(dataType) {
+    (async () => {
+      await new Promise(function(resolve) {
+        Services.clearData.deleteData(dataType, resolve);
+        debug("Data cleared")
```
Comment (okodron) [inline jscomps/PrivateDataManager.js:33]: It's better to print the cleared `dataType` as well.

**Review by okodron — APPROVED** [review]

**Comment by llewelld** [comment]
These look like nice changes to me. Following on from the conversation about whether to combine cookies and site data, I notice that `Services.clearData.deleteData()` already allows the flags to be combined, so if they're both to be cleared together, it may make the code cleaner to combine them into a single call at this stage.
```
this.clearData(Ci.nsIClearDataService.CLEAR_COOKIES | Ci.nsIClearDataService.CLEAR_DOM_STORAGES);
```

**Review by llewelld — APPROVED** [review]
This looks good to me. Splitting "site-data" from "cookies" doesn't look necessary given they're [only ever used together](https://github.com/sailfishos/sailfish-browser/pull/933/files#diff-6ee2a3c1bdc5d546b02a363a605dfbc2fbe3287b0adf62f0208b49cc9a29f5c6R152), but it's fine like this also.

I'd agree with @okodron about adding the type data into the debug message (even if it's shown as a numerical flag). If you have no objections, would you mind adding that? I can then merge.

**Review by okodron — APPROVED** [review]

**Review by llewelld — APPROVED** [review]
The latest changes still look good to me. Thanks Aleksey.

**Pull request merged**